src/server/process_request.c

```diff
@@ -608,9 +611,33 @@ process_request(int sfds)
 		}
 
 		if (isanode == NULL) {
-			if ((acl_check(get_sattr(SVR_ATR_acl_hosts),
-				       request->rq_host, ACL_Host) == 0) &&
-			    (strcasecmp(server_host, request->rq_host) != 0)) {
+			pbs_net_t addr;
+			char ip[PBS_MAXIP + 1];
+
+			addr = get_hostaddr(request->rq_host);
+			if (snprintf(ip, PBS_MAXIP + 1, "%ld.%ld.%ld.%ld",
+			    (addr & 0xff000000) >> 24,
+			    (addr & 0x00ff0000) >> 16,
+			    (addr & 0x0000ff00) >> 8,
+			    (addr & 0x000000ff)) <= 0) {
+				addr = 0;
+			}
+
+			access_allowed = 0;
+
+			if (acl_check(get_sattr(SVR_ATR_acl_hosts), request->rq_host, ACL_Host)) {
```

Comment:
Can we make this acl subnet verification in a separate function and use it irrespective of access_by_krb ?

Reply:
OK. Since we removed the condition `request->rq_type != PBS_BATCH_Connect` above, the variable `access_by_krb` has no meaning anymore. I removed the variable. I also changed the order of evaluation `acl_hosts` and `acl_krb_realms`. `acl_hosts` is first now. It makes more sense. Also, the special case when we access the `server` from the `server` itself is considered here.